src/Components/Body.tsx

```diff
@@ -78,18 +86,75 @@ export function Body(props: State) {
   }
 
   const handleClick = (key, value) => {
+    var dateContent = props.dateTextContents;
+    var words = dateContent[props.index].split(" ");
+    if (key == "day") {
+      resetDate("day", value);
+      words[0] = value + 1;
+    } else if (key == "month") {
+      toggleBox();
+      resetDate("month", value);
+      words[1] = value;
+    } else if (key == "year") {
+      toggleBox();
+      resetDate("year", value);
+      words[2] = value;
+    }
+    dateContent[props.index] =
+      String(words[0]) + " " + String(words[1]) + " " + String(words[2]);
+    props.setDateTextContents([dateContent[0], dateContent[1]]);
+  };
+
+  function handleTextDate(event) {
+    var dateContent = props.dateTextContents;
+    dateContent[props.index] = event;
+    props.setDateTextContents([dateContent[0], dateContent[1]]);
+
+    var errorState = props.dateError;
+    var words = event.split(" ");
+    words[0] = parseInt(words[0]);
+    words[2] = parseInt(words[2]);
+
+    if (

```

Comment:
It looks like you're parsing the date string here, perhaps?  Use a library to do this - I think Date.parse can handle most of these things, for example.